tuist/tuist
# Add `affectsBuiltProduct` flag to skip target scripts when not building for development

### Short description 📝

This pr adds affectsBuiltProduct: Bool to TargetScripts and sets the env var $TUIST_BUILD_FOR_DEVELOPMENT to true in the init for TestService.  Setting the affectsBuiltProduct to true will wrap the TargetScripts build phases with:

if [ ! $TUIST_BUILD_FOR_DEVELOPMENT ]; then
      ...
fi

### How to test the changes locally 🧐

You will need to set the affectsBuiltProduct to true when adding a script to a target.  Then you will need to check to see if that script is ran when running "tuist test".

### Checklist ✅
